### DISCUSSION TITLE
Long training runtime

### DISCUSSION THREAD

**Discussion by IsabelleSajonia** [talmolab/sleap#1459] [Help!] [answered]
I'm using a 5-minute video of a single mouse with 20 labeled frames. In training I picked the single animal option with all of the default settings (plus grayscale) and I'm predicting on 20 random frames. The mean time per epoch is 11:29. Our GPU has 12 GB of memory (GeForce RTX 3060). Our operant chamber that the mouse is in may be on the large size but one wall has levers and the other dispenses food so we can't crop the frame without losing experimental value. Does this seem unusually long, and how many epochs do you recommend training on? Also, would converting/training with grayscale slow it down? I think my predictions might be better without grayscale because with grayscale the mouse's tail blends in with the background.

**Comment by IsabelleSajonia** [comment]
I realized that sleap was not actually accessing my GPU. I fixed it, and my runtime is significantly better! My question about grayscale still stands though.

**Reply by roomrys** [reply, marked as answer]
Hi @IsabelleSajonia,

I am glad you solved your GPU issue. Do you mind following up with how you solved it for posterity?

As for the training time of grayscale versus color...
Although grayscale will have less channels and therefore less features for the same model architecture/parameters, the training time will not be significantly less than color since most of the time will be taken up by processes that do not scale proportionally with number of channels (such as the model architecture and complexity of the problem). 

My assumption would be that if the features are easier to differentiate in color, then you may even have a shorter training time with color (if the model is able to obtain an error plateau in fewer epochs).

Thanks,
Liezl

**Reply by IsabelleSajonia** [reply]
Thanks for the reply! I realized our lab's PC did not have a NVIDA driver, so I downloaded that, and the software is able to access the GPU now. Sometimes it will slow down again and when I check the system summary, I see that it is not detecting a GPU. Rebooting usually fixes the issue but I was wondering if you'd recommend an alternative GPU?

**Reply by IsabelleSajonia** [reply]
And your grayscale answer is very helpful thank you!

**Reply by roomrys** [reply]
Hi @IsabelleSajonia,

I have the same GPU and it works well for me.... The intermittent connection sounds annoying. It could be a problem with the driver version (I assume you are on the latest, so trying to roll back a few months to troubleshoot) or a bad physical connection to your PCIe?

Thanks,
Liezl

**Reply by IsabelleSajonia** [reply]
Hm ok I'll try to troubleshoot that! I know this is a different topic but I also had a question about video quality. When I import videos into sleap to label I noticed that the quality is worse than playing them outside of the software. Our resolution is 1024x768 and frame rate is 24.99 (mp4). Do you have any idea why that might be or how I can improve that without increasing the resolution since I have reached my labs maximum? Thanks so much for helping me get started!

**Reply by roomrys** [reply]
Hi @IsabelleSajonia,

In the Videos table there should be column for height and width of the video. Do those match your original video resolution? Or are you referring to a change in frame rate? SLEAP should not adjust the video (other than the option for color/grayscale). 

If the observation is qualitative, could it be because you are noticing an up-sampling of the video to fit to the window (and allow more precise labeling)?

Thanks,
Liezl

**Reply by IsabelleSajonia** [reply]
Yes the height and width in the table match my original video so I think it is just qualitative. To me this looks harder to label (more pixelated).

**Reply by roomrys** [reply]
Idea created! #1471
